### PR TITLE
805 806 807 Fix week available for offer

### PR DESCRIPTION
**Pull request by pierrax**

### CODE REVIEW

**Review by fitchMitch — CHANGES_REQUESTED** [review]

File: app/controllers/dashboard/internship_offers_controller.rb [hunk 1]
```diff
@@ -32,21 +32,21 @@ def create
         end
         on.failure do |failed_internship_offer|
           @internship_offer = failed_internship_offer || InternshipOffer.new
-          @available_weeks = Week.selectable_from_now_until_end_of_school_year
+          @available_weeks = Week.selectable_on_school_year
```
[the remark this answers is not in the document]
Comment (pierrax) [reply app/controllers/dashboard/internship_offers_controller.rb]: yes fixed

File: app/controllers/dashboard/internship_offers_controller.rb [hunk 2]
```diff
@@ -32,21 +32,21 @@ def create
...
+      @available_weeks = Week.selectable_on_school_year
       render :new, status: :bad_request
     end
 
     def edit
       authorize! :update, @internship_offer
       @republish = params[:republish].present?
       @republish = true
-      @available_weeks = @internship_offer.available_weeks_when_editing
+      @available_weeks = Week.selectable_on_school_year
```
[the remark this answers is not in the document]
Comment (pierrax) [reply app/controllers/dashboard/internship_offers_controller.rb]: yes fixed

File: app/models/concerns/stepper_proxy/internship_offer_info.rb [hunk 3]
```diff
@@ -78,14 +78,8 @@ def available_weeks
     def available_weeks_when_editing
       raise 'Wrong !!' unless persisted? && respond_to?(:weeks) # temporary
```
[the remark this answers is not in the document]
Comment (pierrax) [reply app/models/concerns/stepper_proxy/internship_offer_info.rb]: yes fixed

File: app/controllers/dashboard/internship_offers_controller.rb [hunk 4]
```diff
@@ -32,21 +32,21 @@ def create
...
           @internship_offer = failed_internship_offer || InternshipOffer.new
-          @available_weeks = Week.selectable_from_now_until_end_of_school_year
+          @available_weeks = Week.selectable_on_school_year
           render :new, status: :bad_request
         end
       end
     rescue ActionController::ParameterMissing
       @internship_offer = InternshipOffer.new
-      @available_weeks = Week.selectable_from_now_until_end_of_school_year
+      @available_weeks = Week.selectable_on_school_year
```
[the remark this answers is not in the document]
Comment (pierrax) [reply app/controllers/dashboard/internship_offers_controller.rb]: yes fixed

**Pull request closed without merging**